src/common/include/gudhi/graph_simplicial_complex.h

```diff
@@ -30,40 +31,127 @@ struct vertex_filtration_t {
   typedef boost::vertex_property_tag kind;
 };
 
-/** \brief Proximity_graph contains the vertices and edges with their filtration values in order to store the result
- * of `Gudhi::compute_proximity_graph` function.
+/** \brief Filtered_edges_vector contains the edges with their filtration values, sorted by ascending filtration values,
+ * in order to store the result of `Gudhi::compute_edge_graph` function.
  *
- * \tparam SimplicialComplexForProximityGraph furnishes `Filtration_value` type definition.
+ * \tparam SimplicialComplexForFilteredEdges furnishes `Filtration_value` and `Vertex_handle` types definitions.
  *
  */
-template <typename SimplicialComplexForProximityGraph>
-using Proximity_graph = typename boost::adjacency_list < boost::vecS, boost::vecS, boost::directedS
-, boost::property < vertex_filtration_t, typename SimplicialComplexForProximityGraph::Filtration_value >
-, boost::property < edge_filtration_t, typename SimplicialComplexForProximityGraph::Filtration_value >>;
+template <typename SimplicialComplexForFilteredEdges>
+class Filtered_edges_vector {
+ public:
+  /** \brief Type of the `Gudhi::Filtered_edges_vector` data structure.
+   */
+  using Filtered_edge = std::tuple<typename SimplicialComplexForFilteredEdges::Filtration_value,
+                                   typename SimplicialComplexForFilteredEdges::Vertex_handle,
+                                   typename SimplicialComplexForFilteredEdges::Vertex_handle>;
+  using Filtered_edge_set = std::vector<Filtered_edge>;
+
+  template <class EdgeIterator, class EdgePropertyIterator>
+  Filtered_edges_vector(EdgeIterator first, EdgeIterator last, EdgePropertyIterator ep_iter,
+                        typename SimplicialComplexForFilteredEdges::Vertex_handle n) {
+    for (; first < last; ++first) {
+      edges_.push_back(std::make_tuple(*ep_iter, first->first, first->second));
+      ++ep_iter;
+    }
+    // By default the sort is done on the first element (Filtration_value) in ascending order.
+    std::sort(edges_.begin(), edges_.end());
+  }
 
-/** \brief Computes the proximity graph of the points.
+  Filtered_edge_set edges() const { return edges_; }
+
+  /** \brief Returns the edges vector at a specific index filtration value.
```

Comment:
It doesn't look like this returns a vector... Returns the filtration value of the n-th edge? (the edge of index n?)